[Improve][Connector-V2][Http] Supports Cursor-based Pagination (#9109) (#9138)
apache/seatunnel: docs/en/connector-v2/source/Http.md

| pageing.page_field            | String  | No       | -                    | This parameter is used to specify the page field name in the request parameter                                                                                                    |
| pageing.total_page_size       | Int     | No       | -                    | This parameter is used to control the total number of pages                                                                                                                       |
| pageing.batch_size            | Int     | No       | -                    | The batch size returned per request is used to determine whether to continue when the total number of pages is unknown                                                            |
| pageing.start_page_number     | Int     | No       | 1                    | Specify the page number from which synchronization starts                                                                                                                         |

> | pageing.page_type             | String  | No       | PageNumberPagination | this parameter is used to specify the page type ,or PageNumberPagination if not set, only support `PageNumberPagination` and `CursorPagination`.                                  |

Reviewer: ```suggestion
| pageing.page_type             | String  | No       | PageNumber | this parameter is used to specify the page type ,or PageNumber if not set, only support `PageNumber` and `Cursor`.                                  |
```

- See this link for task configuration [http_jsonpath_to_assert.conf](../../../../seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/resources/http_jsonpath_to_assert.conf).

### pageing

> The current supported pagination type are `PageNumberPagination` and `CursorPagination`. 

Reviewer: ```suggestion
The current supported pagination type are `PageNumber` and `Cursor`. 
```

> if you need to use pagination, you need to configure `pageing`. the default pagination type is `PageNumberPagination`.

Reviewer: ```suggestion
if you need to use pagination, you need to configure `pageing`. the default pagination type is `PageNumber`.
```

> #### 1. PageNumberPagination

Reviewer: ```suggestion
#### PageNumber
```

}

``` 

> #### 2. CursorPagination

Reviewer: ```suggestion
#### Cursor
```

the `pageing.page_type` parameter must be set to `CursorPagination`.

> `cursor_field` is the field name of the cursor  in the request parameters.

Reviewer: ```suggestion
`cursor_field` is the field name of the cursor in the request parameters.
```

`cursor_response_field` is the field name of the cursor field in the response data, this value is then passed as the value of the `cursor_field` parameter in the next request.
````hocon

> source {

Reviewer: we should add this https://github.com/apache/seatunnel/pull/9138/files#diff-31bd0672f7a316bde3b591f194484526bed84f71b007370b75c2c2bf99223fb4R386-R388 to add pageing fields into request.

Author: Did you mean that the explanation above wasn't clear enough， and we should explicitly state than the value of this parameter  must be passed in as a pagination field request parameter ? (You mean the phrase 'add pageing fields into request')
